### PR TITLE
[WIP] Atom Convolutions Into TensorGraph

### PR DESCRIPTION
**Pull request by lilleswing**
Atom Convolutions Into Tensorgraph.

Does not have latest enhancements yet of batch norm or per atom dense layers.

### CODE REVIEW

**Comment by lilleswing** [comment]
@joegomes was there a reason you didn't flip the flag for allowing back propagation through the per atom level dense networks?

https://www.tensorflow.org/api_docs/python/tf/map_fn

File: deepchem/feat/atomic_coordinates.py [hunk 1]
```diff
@@ -10,80 +10,170 @@
...
 import numpy as np
-import mdtraj
+from rdkit import Chem
 from deepchem.feat import Featurizer
 from deepchem.feat import ComplexFeaturizer
-from deepchem.utils import rdkit_util
+from deepchem.utils import pad_array, rdkit_util
 
 
-class AtomicCoordinates(Featurizer):
```
Comment (rbharath) [inline deepchem/feat/atomic_coordinates.py]: The atomic-convs code seems to use the old neighbor-listing implementation I did in pure python instead of using `mdtraj`'s optimized version that @peastman added in a few PRs ago. Might be worth just making the change to use the `mdtraj` implementation which is much faster rather than my custom implementation.

File: deepchem/feat/tests/test_atomic_coordinates.py [hunk 2]
```diff
@@ -116,37 +118,18 @@ def test_neighbor_list_max_num_neighbors(self):
       else:
         assert nblist[i] == []
 
-  def test_neighbor_list_periodic(self):
```
Comment (rbharath) [inline deepchem/feat/tests/test_atomic_coordinates.py]: We probably shouldn't remove this test.

File: examples/pdbbind/pdbbind_atomic_conv.py [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+  frag2_conv = frag1_conv.shared(in_layers=[frag2_conv])
+  complex_conv = frag1_conv.shared(in_layers=[complex_conv])
+
+score = AtomicConvScore(in_layers=[frag1_conv, frag2_conv, complex_conv])
+
+label = Label(shape=(None, 1))
+loss = L2LossLayer(in_layers=[score, label])
+
+
+def feed_dict_generator(dataset, batch_size, epochs=1):
```
Comment (rbharath) [inline examples/pdbbind/pdbbind_atomic_conv.py]: This is pretty complicated :). Let's brainstorm ways in future PRs to make constructing feed-dicts easier for complex models.

File: deepchem/trans/transformers.py [hunk 4]
```diff
@@ -895,3 +895,27 @@ def UG_to_DAG(self, sample):
...
+
+  def __init__(self,
+               size,
+               transform_X=True,
+               transform_y=False,
+               transform_w=False):
+    """Initializes transformation based on dataset statistics."""
+    self.size = size
+    self.transform_X=True
+    self.transofrm_y=False
```
Comment (proteneer) [inline deepchem/trans/transformers.py]: typo

**Comment by lilleswing** [comment]
Closed in favor of
https://github.com/deepchem/deepchem/pull/530
